resolver-dns/src/main/java/io/netty/resolver/dns/DnsNameResolverContext.java

```diff
@@ -436,7 +481,10 @@ private void finishResolve() {
     }
 
     protected abstract boolean finishResolve(
-            Class<? extends InetAddress> addressType, List<DnsCacheEntry> resolvedEntries);
+            Class<? extends InetAddress> addressType, List<DnsCacheEntry> resolvedEntries, Promise<T> promise);
+
+    protected abstract DnsNameResolverContext<T> newResolverContext(DnsNameResolver parent, String hostname,
```

Comment:
this was made protected because the abstract `finishResolve` method was `protected` - I can make it package private, however then it would make sense to do the same for `finishResolve`, WDYT ?